### PR TITLE
Change the Whitehall tagging interface to respect Phases

### PR DESCRIPTION
**Pull request by koetsier**
The Whitehall tagging interface differentiated between Published and Draft taxons, to show that some were shown on GOV.UK and some (specifically Corporate Information) were not, while allowing publishers to tag to both types

Now that all taxons have been published and we're instead using Alpha, Beta and Live, the current interface should be updated to show which Taxons are Live and which ones are in Alpha/Beta

Trello: https://trello.com/c/lGAGierw/183-change-the-whitehall-tagging-interface-to-respect-phases

### CODE REVIEW

**Comment by koetsier** [comment]
Use: `rake taxonomy:rebuild_cache` to populate Redis with the taxonomy trees

File: lib/taxonomy/tree.rb [hunk 1]
```diff
@@ -10,30 +10,33 @@ def initialize(expanded_root_taxon_hash)
       @root_taxon = build_taxon(expanded_root_taxon_hash)
       root_taxon.children = parse_taxons(
         root_taxon,
-        expanded_root_taxon_hash['expanded_links_hash']
+          expanded_root_taxon_hash['expanded_links_hash']
```
Comment (cbaines) [inline lib/taxonomy/tree.rb]: This indentation change looks unnecessary.

File: lib/taxonomy/taxon.rb [hunk 2]
```diff
@@ -3,25 +3,27 @@
...
       @base_path = base_path
+      @phase = phase
+      @visible_to_departmental_editors = visible_to_departmental_editors
       @children = []
     end
 
-    def tree
-      children.each_with_object([self]) do |child, tree|
-        tree.concat(child.tree)
+    def taxon_list
```
Comment (cbaines) [inline lib/taxonomy/taxon.rb]: Tricky naming issue, but I think this is an improvement.

File: lib/taxonomy/tree.rb [hunk 3]
```diff
@@ -10,30 +10,33 @@ def initialize(expanded_root_taxon_hash)
...
     end
 
-    def child_nodes(item_hash, key)
-      item_hash
-        .fetch(key, {})
-        .fetch('child_taxons', [])
-        .sort_by { |hsh| hsh.fetch('title') }
+    def child_nodes(item_hash)
+      children = item_hash.dig('links', 'child_taxons') ||
+        item_hash.dig('expanded_links', 'child_taxons') || []
```
Comment (cbaines) [inline lib/taxonomy/tree.rb]: It looks like the previous `key` parameter was never used, so it's probably safe to assume that only the expanded links will ever exist.

**Pull request merged**